### ISSUE TITLE
[Bug] KroneckerMultiTaskGP posterior fails for training_inputs with more than 800 rows

### ISSUE THREAD

**Issue by fleskovar** [pytorch/botorch#956]
# 🐛 Bug

When conditioning KroneckerMultiTaskGP with training inputs that have more than 800 rows, the function that calculates the posterior fails. This is happening because  the diagonalization of  `data_data_covar` is done with `lanczos` instead of `symeig`, resulting in shape of `data_data_evals` to be different than `train_x.shape[0]`. This difference in shapes causes `train_noise` to have the wrong shape, making the function fail when trying to calculate `full_data_inv_evals`.

## To reproduce

** Code snippet to reproduce **
```python
from botorch.models.multitask import KroneckerMultiTaskGP
import torch

train_X = torch.rand(801, 2)
train_Y = torch.cat([train_X, train_X**2], dim=-1)
model = KroneckerMultiTaskGP(train_X, train_Y)
model.posterior(torch.tensor([[0.5, 0.5]]))
```

** Stack trace/error message **
```
Traceback (most recent call last):
  File "c:\Users\asd\Documents\GitHub\test\refactor\consolidated\test.py", line 7, in <module>
    model.posterior(torch.tensor([[0.5, 0.5]]))
  File "C:\Users\asd\miniconda3\envs\test\lib\site-packages\botorch\models\multitask.py", line 621, in posterior
    KroneckerProductDiagLazyTensor(
  File "c:\users\asd\documents\github\gpytorch\gpytorch\lazy\diag_lazy_tensor.py", line 31, in __add__
    return self.add_diag(other._diag)
  File "c:\users\asd\documents\github\gpytorch\gpytorch\lazy\diag_lazy_tensor.py", line 105, in add_diag
    shape = _mul_broadcast_shape(self._diag.shape, added_diag.shape)
  File "c:\users\asd\documents\github\gpytorch\gpytorch\utils\broadcasting.py", line 20, in _mul_broadcast_shape
    raise RuntimeError("Shapes are not broadcastable for mul operation")
RuntimeError: Shapes are not broadcastable for mul operation
```

## Additional context

Changing the `train_noise` calculation to the code below seems to solve the issue, but I am not sure if there are other parts in the code where `data_data_evals` shape is assumed to be the same as `train_x.shape[0]`

```python
train_noise = self.likelihood._shaped_noise_covar(torch.Size([data_data_evals.shape[0] ,*train_x.shape[1:]]))
```

**Comment by wjmaddox** [comment]
Ugh, sorry for the late response. This is a bug due to what happens on the gpytorch end as we move from cholesky / symeig linear algebra into using lanczos. 

A simple workaround for now is:
```python
with gpytorch.settings.max_cholesky_size(10000):
    model.posterior(torch.tensor([[0.5, 0.5]]))
```
which enforces cholesky / symeig linear algebra.

I'll put up a PR resolving this hopefully soon.

**Comment by saitcakmak** [comment]
I believe this was fixed by #1071. Thanks @wjmaddox !
